src/web_video_server.cpp

```diff
@@ -102,33 +104,26 @@ WebVideoServer::WebVideoServer(rclcpp::Node::SharedPtr & node)
     );
   } catch (boost::exception & e) {
     RCLCPP_ERROR(
-      node_->get_logger(), "Exception when creating the web server! %s:%d",
+      get_logger(), "Exception when creating the web server! %s:%d",
       address_.c_str(), port_);
     throw;
   }
+  RCLCPP_INFO(get_logger(), "Waiting For connections on %s:%d", address_.c_str(), port_);
+  if (publish_rate_ > 0) {
+    create_wall_timer(1s / publish_rate_, [this]() {restreamFrames(1.0 / publish_rate_);});
+  }
+  server_->run();
```

Comment:
Unless I'm missing something, we should be able to get rid of `setup_cleanup_inactive_streams()` method and just create the timer in the constuctor. That way, the timer will created when spawning the component in component container.

```suggestion
  
  cleanup_timer_ = create_wall_timer(500ms, [this]() {cleanup_inactive_streams();});

  server_->run();
```

Also the type for `cleanup_timer_` should be changed to `rclcpp::TimerBase::SharedPtr`